### DISCUSSION TITLE
Maven repo dependency url is taking "forever"

### DISCUSSION THREAD

**Discussion by jactor-rises** [community/community#25926] [Actions]
<p>I have a repository on GitHub which had GitHub Actions since beta release of GitHub Actions without any troubles.</p>
<p>This is a simple java-component without lots of dependencies and since it is used during testing of other components, it is really fast to build (about 4 seconds without downloading dependencies)</p>
<p>But this changed yesterday when I bumped spring-boot dependency 2.3.2.RELEASE. It took over 4 hours and 20 minutes to finish (!!!): <a href="https://github.com/navikt/bidrag-commons-test/pull/49/checks?check_run_id=919269320" rel="nofollow noopener">https://github.com/navikt/bidrag-commons-test/pull/49/checks?check_run_id=919269320</a></p>
<p>This was a feature branch which is merged to master this morning and it is still building…</p>
<p>It seems that it first is using our in-house maven repo, before trying the central maven repo… I do not know why, this is a github runner using ubuntu-latest and I cannot figure out where it can get the url from…</p>

**Comment by jactor-rises** [comment]
<p>when it executes <code>mvn -B install -e</code> it downloaded a bunch of dependencies from spring-boot, but this is done in 10 seconds (including the build)…</p>
<p>an in-house action causes the download-anarchy … this is essiantal a bash-script and I do not understand…</p>
<p>bash script:</p>
<pre><code class="lang-auto">#!/bin/bash
set -e

if [ ! -z "$INPUT_SRC_FOLDER" ]
then
  cd "$INPUT_SRC_FOLDER"
fi

echo "Working directory"
pwd

if [ -z "$INPUT_MAVEN_IMAGE" ]; then
  mvn -B dependency:tree | tee .dependency-tree
else
  docker run --rm -v "$PWD":/usr/src/mymaven -v "$HOME"/.m2:/root/.m2 -w /usr/src/mymaven \
    "$INPUT_MAVEN_IMAGE" mvn -B dependency:tree | tee .dependency-tree
fi

cat .dependency-tree | grep BUILD | grep -c SUCCESS | sed 's/1//' # fails if count is 0

DEPENDENCIES=$(cat .dependency-tree | grep "\[INFO]" | grep -e "\- " -e "+-")
COUNT=$(echo "$DEPENDENCIES" | grep -c SNAPSHOT || true)

echo "Found $COUNT SNAPSHOT dependencies"

if [ "$COUNT" -gt 0 ]
  then
    &gt;&amp;2 echo ::error No SNAPSHOT dependencies allowed
    exit 1;
fi
</code></pre>

**Comment by weide-zhou** [comment, marked as answer]
<p>Hi <a class="mention" href="/jactor-rises">@jactor-rises</a>,</p>
<p>This is due to it takes more time to download from local-releases now: <a href="https://repo.adeo.no/repository/maven-releases" rel="nofollow noopener">https://repo.adeo.no/repository/maven-releases</a>.  I tried to ping the url but failed, it seems not accessable now.<br>
In your <a href="https://github.com/navikt/bidrag-commons-test/blob/1de0943dc6d3604600fdc61111b2592e09e6dc89/pom.xml#L83" rel="nofollow noopener">pom.xml</a> setting, please comment the section as below:<br>
<img src="https://user-images.githubusercontent.com/52772761/181094153-e9ee31d6-2f14-4bff-948e-6afb93a3fc88.png" alt="image" data-base62-sha1="meSd67jFVJaX39tIWoBzxAzUbj" width="493" height="185"></p>
<p>It will only download from central now, and it only takes 15s on my fork repo.<br>
Please refer to my <a href="https://github.com/weide-zhou/bidrag-commons-test/runs/925955632?check_suite_focus=true#step:5:9" rel="nofollow noopener">workflow</a> for more details.</p>
<p>Thanks.</p>

**Comment by jactor-rises** [comment]
<p>I am so sorry for not seeing that… I was blind for that error since this has always been a public open repository not needing the internal maven repository</p>
<p>Thanx for the effort</p>

**Comment by weide-zhou** [comment]
<p>Hi <a class="mention" href="/jactor-rises">@jactor-rises</a>,</p>
<p>Thanks for your reply! Is there any other questions left in this ticket i can help with?</p>

**Comment by jactor-rises** [comment]
<p>nothing… I am embarrassed that I didn’t notice this earlier… :roll_eyes:</p>

**Comment by weide-zhou** [comment]
<p>Thanks for your confirmation! <a class="mention" href="/jactor-rises">@jactor-rises</a> . Glad to support!</p>
<p>Happy coding!</p>
